internal/controller/cluster_cleanup.go

```diff
@@ -44,8 +45,23 @@ func (r *ClusterReconciler) cleanupCompletedJobs(
 			continue
 		}
 
+		contextLogger.Debug("Removing finalizer from job", "job", job.Name)
```

Comment:
I would move the logger just before the patch, otherwise if `removeFinalizer` returns false you are logging anyway.